docusaurus/docs/protocol/primitives/claim_and_proof_lifecycle.md

```diff
@@ -80,6 +79,53 @@ sequenceDiagram
+|------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
+| `num_blocks_per_session`           | The number of blocks between the session start & end heights. Relays handled in these blocks are included in session N. It is positively correlated with the number of relays in (i.e. size of) each session tree for each session number (less other scaling factors; e.g. relaymining).                                                                                                        | 4       |
+| `grace_period_end_offset_blocks`   | The number of blocks after the session end height, at which the grace period ends. Valid relays from both sessions N and N +1 are accepted in these blocks. It is positively correlated to the amount of time gateways have to transition sending relays to suppliers in the next session.                                                                                                       | 1       |
+| `claim_window_open_offset_blocks`  | The number of blocks after the session grace period height, at which the claim window opens. Valid claims for session N will be rejected in these blocks. This parameter MUST NOT be less than grace_period_end_offset_blocks. It is positively correlated with the number of relays in (i.e. size of) each session tree for each session number (less other scaling factors; e.g. relaymining). | 1       |
+| `claim_window_close_offset_blocks` | The number of blocks after the claim window open height, at which the claim window closes. Valid claims for session N will be accepted in these blocks. It is negatively correlated with density of claim creation (and update) messages over blocks in a given session number.                                                                                                                  | 4       |
+| `proof_window_open_offset_blocks`  | The number of blocks after the claim window close height, at which the proof window opens. Valid proofs for session N will be rejected in these block. It is positively correlated with the amount of time suppliers MUST persist the complete merkle trees for unproven sessions (proof path is revealed at proof_window_open_height - 1).                                                      | 0       |
```

Comment:
Maybe add a TOTO to make proof path to be revealed at earliest supplier commit height - 1

Reply:
I thought the idea was to reveal it as late as possible to reduce the duration between when it can be known and when it must be used?

Reply:
If it is revealed within the claim window (i.e. before claim window close), that's even worse as the proof path would be known before the claim (commitment) is on-chain. Am I missing something?

Reply:
Added the comment w.r.t. this
> (proof path is revealed at proof_window_open_height - 1).

Reply:
Ah, I see it now. :dizzy_face: 

```suggestion
| `proof_window_open_offset_blocks`  | The number of blocks after the claim window close height, at which the proof window opens. Valid proofs for session N will be rejected in these block. It is positively correlated with the amount of time suppliers MUST persist the complete merkle trees for unproven sessions (proof path is revealed at earliest_supplier_proof_commit_height - 1).                                                      | 0       |
```